include/oneapi/dpl/internal/dynamic_selection_impl/dynamic_load_policy.h

```diff
@@ -0,0 +1,170 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_DYNAMIC_LOAD_POLICY_H
+#define _ONEDPL_DYNAMIC_LOAD_POLICY_H
+
+#include <atomic>
+#include <memory>
+#include <limits>
+#include <mutex>
+#include <vector>
+#include <exception>
+#include <type_traits>
+#include <utility>
+#include "oneapi/dpl/internal/dynamic_selection_traits.h"
+#if _DS_BACKEND_SYCL != 0
+    #include "oneapi/dpl/internal/dynamic_selection_impl/sycl_backend.h"
+#endif
+
+namespace oneapi {
+namespace dpl{
+namespace experimental{
+
+#if _DS_BACKEND_SYCL != 0
+  template <typename Backend=sycl_backend>
+#else
+  template <typename Backend>
+#endif
+  struct dynamic_load_policy {
+    private:
+    using backend_t = Backend;
+    using load_t = int;
+    using execution_resource_t = typename backend_t::execution_resource_t;
+
+    public:
+    //Policy Traits
+    using resource_type = typename backend_t::resource_type;
+    using wait_type = typename backend_t::wait_type;
+
+    //private:
+    struct resource_t {
+      execution_resource_t e_;
+      std::atomic<load_t> load_;
+      resource_t(execution_resource_t e) : e_(e) {load_=0;}
+    };
+    using resource_container_t = std::vector<std::shared_ptr<resource_t>>;
+
+    template<typename Policy>
+    class dl_selection_handle_t {
+      Policy policy_;
+      std::shared_ptr<resource_t> resource_;
+
+    public:
+      dl_selection_handle_t(const Policy& p,std::shared_ptr<resource_t> r )
+        : policy_(p), resource_(r) {}
+
+      auto unwrap() { return ::oneapi::dpl::experimental::unwrap(resource_->e_); }
+
+      Policy get_policy() { return policy_; };
+
+      void report(const execution_info::task_submission_t&) {
+        resource_->load_.fetch_add(1);
+      }
+
+      void report(const execution_info::task_completion_t&) const  {
+        resource_->load_.fetch_sub(1);
+      }
+    };
+
+    std::shared_ptr<backend_t> backend_;
+
+    using selection_type = dl_selection_handle_t<dynamic_load_policy<Backend>>;
+
+    struct state_t{
+        resource_container_t resources_;
+        int offset;
```

Comment:
Better to do initialization right here: ```int offset = 0;```

Reply:
Offset is not a property of dynamic_load_policy anymore.